demux/demux.c

```diff
@@ -2942,6 +2942,19 @@ static struct replaygain_data *decode_rgain(struct mp_log *log,
         return talloc_dup(NULL, &rg);
     }
 
+    // The r128 replaygain tags declared in RFC 7845 for opus files. The tags
+    // are generated with EBU-R128, which does not use peak meters. And the
+    // values are stored as a Q7.8 fixed point number in dB.
+    if (decode_gain(log, tags, "R128_TRACK_GAIN", &rg.track_gain) >= 0) {
+        if (decode_gain(log, tags, "R128_ALBUM_GAIN", &rg.album_gain) < 0) {
```

Comment:
Looking at the [spec](https://datatracker.ietf.org/doc/html/rfc7845.html#page-25), it doesn't seem like to me that `R128_TRACK_GAIN` or `R128_ALBUM_GAIN` are dependent on one another. I think they should both be checked independent of one another.

Reply:
The using of nested if statements is to be consistent with the above replaygain tag handling code (at line [2924](https://github.com/datasone/mpv/blob/354ba9412b8b0f71cef077fb05487cb200124a5c/demux/demux.c#L2924)). Besides, I think separating the tag decode operations would actually complicate the code: the replaygain data (`rg.track_gain`) would then need to be validated again before returning, as the returned value is further used to indicate the validality [here](https://github.com/datasone/mpv/blob/354ba9412b8b0f71cef077fb05487cb200124a5c/demux/demux.c#L2967).

Reply:
Yeah it looks like `export_replaygain` uses this same assumption so even if you split it out, it wouldn't work without the track gain being defined. I guess there's no point in trying to refactor this now so LGTM. I'll test it out later.